### PR TITLE
embroider app tests & use @embroider as peerDeps

### PR DESCRIPTION
**Pull request by AmauryD**
Fixes https://github.com/kategengler/ember-cli-code-coverage/issues/374 by using the app installed @embroider/compat. 

Note: if you want i can help moving from yarn 1 to pnpm. It was a nightmare to deal with the dependencies of the test-packages.

### CODE REVIEW

**Comment by rohitpaulk** [comment]
Just tested this out in our app, and it does seem to fix https://github.com/kategengler/ember-cli-code-coverage/issues/374.

I was able to workaround the sub-directory github link limitation mentioned in https://github.com/kategengler/ember-cli-code-coverage/issues/374#issuecomment-1556788031 by using https://gitpkg.vercel.app/:

```json
"ember-cli-code-coverage": "https://gitpkg.now.sh/AmauryD/ember-cli-code-coverage/packages/ember-cli-code-coverage?master"
```

**Comment by elidupuis** [comment]
I tested this out in our app (using the `gitpkg.now.sh` approach [outlined above](https://github.com/kategengler/ember-cli-code-coverage/pull/382#issuecomment-1558314297)) and can also confirm that it works as expected. For reference, I also upgraded Embroider the latest 3.x versions [outlined here](https://github.com/embroider-build/embroider/releases/tag/v3.1.0-addon-dev).

File: packages/ember-cli-code-coverage/package.json [hunk 1]
```diff
@@ -91,6 +89,18 @@
     "release-it-lerna-changelog": "^2.3.0",
     "sinon": "^4.2.2"
   },
+  "peerDependencies": {
+    "@embroider/compat": "^0.47.0 || ^1.0.0 || ^2.0.0 || ^3.0.0",
```
Comment (kategengler) [inline packages/ember-cli-code-coverage/package.json:93]: Can you loosen these to make the last `>=3.0.0`? That way if it happens to work with 4.0 or further, we don't have to release just to bump the peer dep?

File: test-packages/.DS_Store
Comment (kategengler) [inline test-packages/.DS_Store:1]: I think this is an accidentally committed file

**Review by kategengler — CHANGES_REQUESTED** [review]

File: test-packages/.DS_Store
Comment (AmauryD) [reply test-packages/.DS_Store:1]: yes, it is. I added .DS_Store files to gitignore

File: packages/ember-cli-code-coverage/package.json (hunk 1 above)
Comment (AmauryD) [reply packages/ember-cli-code-coverage/package.json]: done ! Thank's for the review

File: .DS_Store
Comment (kategengler) [inline .DS_Store:1]: Seems like another DS_Store made it in

Comment (AmauryD) [reply .DS_Store:1]: Done ! This was the last one. 

image

**Comment by RobbieTheWagner** [comment]
> Fixes #374 by using the app installed @embroider/compat.
> 
> Note: if you want i can help moving from yarn 1 to pnpm. It was a nightmare to deal with the dependencies of the test-packages.

@AmauryD if the offer still stands, I would love help converting to pnpm. It seems the tests you added are failing in the master branch now, and I am not sure why. Also to note, #387 makes some embroider tweaks and I think we should probably get that in too. Would love your help getting this stuff sorted out and getting tests passing again :heart:

**Comment by AmauryD** [comment]
> > Fixes #374 by using the app installed @embroider/compat.
> > Note: if you want i can help moving from yarn 1 to pnpm. It was a nightmare to deal with the dependencies of the test-packages.
> 
> @AmauryD if the offer still stands, I would love help converting to pnpm. It seems the tests you added are failing in the master branch now, and I am not sure why. Also to note, #387 makes some embroider tweaks and I think we should probably get that in too. Would love your help getting this stuff sorted out and getting tests passing again ❤️

It will be a pleasure to help ! I'll take a look in my free time 😄

**Comment by RobbieTheWagner** [comment]
Thank you @AmauryD!

**Pull request merged**